process/block/preprocess/basePreProcess.go

```diff
@@ -307,3 +312,47 @@ func (bpp *basePreProcess) computeGasConsumedByTx(
 
 	return txGasLimitInSenderShard, txGasLimitInReceiverShard, nil
 }
+
+func (bpp *basePreProcess) saveAccountBalanceForAddress(address []byte) {
+	if bpp.balanceComputation.HasAddressBalanceSet(address) {
+		return
+	}
+
+	balance, err := bpp.getBalanceForAddress(address)
+	if err != nil {
+		//TODO: Analyze if it is better to call return here
```

Comment:
I think this is ok. If there is an error here (because the account does not exists, (adress conversion will not raise error & type assertion should not occur) ) the account is snapshot with balance 0 meaning there is no possibility of a debit tx occurring withing this block.

Reply:
Yep, this is how I thought about. Removed the TODO